src/driver/mongodb/MongoQueryRunner.ts

```diff
@@ -545,24 +580,57 @@ export class MongoQueryRunner implements QueryRunner {
     }
 
     /**
-     * Starts transaction.
+     * Starts transaction. Mongo allows transactions happen out of depth order but for simplification we will use
+     * it like all the others
+     * Read more: https://www.mongodb.com/docs/manual/core/transactions/
      */
     async startTransaction(): Promise<void> {
-        // transactions are not supported by mongodb driver, so simply don't do anything here
+        this.isTransactionActive = true
+        this.transactionDepth += 1
+        const session = this.databaseConnection.startSession(
+            this.sessionOptions,
+        )
+        session.startTransaction(this.transactionOptions)
+
+        this.sessions[this.transactionDepth] = session
     }
 
     /**
      * Commits transaction.
      */
     async commitTransaction(): Promise<void> {
-        // transactions are not supported by mongodb driver, so simply don't do anything here
+        if (!this.isTransactionActive) throw new TransactionNotStartedError()
```

Comment:
nit: Is this variable necessary given you can infer it from `transactionDepth`?

I know this is more expressive, but it adds an extra place where a bug may happen because it was not flipped and could be refactored into a function based on `transactionDepth`